### PR TITLE
Refactor/activity class

### PR DESCRIPTION
**Pull request by elizhahn**
## Purpose
_Describe the problem or feature(s) being addressed._
- Creating functionality for the activity class and removing activity related methods/properties from the user class
## Abstract
_How does this change address the problem? Give a brief summary of the change. _
- The user class was currently running almost all of the methods for the Activity class. The methods/properties were taken from that class and placed into Activity class
- Activity class was made a child class of the User class so it could access properties that were needed from the use class such as strikeLength and daily step goal, date, and id
#### Pre-Merge TODOs
- [x] run npm test when in the root director, all files should be passing
- [x] Please think of possible sad path tests for the future and comment with ideas
- [x] Check Styling/Syntax
- [x] Check for opportunities for further refactoring

## Learning
_Describe the research stage, any interesting findings or resources to note?_
- To implement the child class, importing was a barrier. Further research is needed to see if we can instantiate a child class within a parent class and import those files in a way that makes that possible
#### Next Steps
_What needs to be worked on still in this feature/fix?_
- sad path testing
- potentially a method to return an object that holds a weekly snapshot of  step count and minutes active for the past 7 recorded days

### CODE REVIEW

File: test/Activity-test.js [hunk 1]
```diff
@@ -22,76 +21,123 @@ describe('Activity', function() {
...
-    expect(user1.activityRecord.length).to.equal(1);
+  it('should have a default activity record', function() {
+    expect(activity1.activityRecord).to.deep.equal([]);
   });
-  it('should have a method that calculate miles walked', function() {
-    expect(activity1.calculateMiles(userRepository)).to.equal('3.0');
-  });
-  describe('compareStepGoal', function() {
+  describe('Activity Methods', function () {
+    it('should find a user\'s steps, minutes active, and flights climbed', function() {
```
Comment (ConnorAndersonLarson) [inline test/Activity-test.js:58]: SAD path: What if no user id is found? What should be returned?

File: src/Activity.js [hunk 2]
```diff
@@ -1,31 +1,82 @@
...
+    const averageWeeklyData = currentWeekData.reduce((weeklyAverage, day) => {
+      if(!weeklyAverage.steps) {
+       weeklyAverage.steps = day.numSteps;
+       weeklyAverage.minutesActive = day.minutesActive; 
+    } else {
+       weeklyAverage.steps += day.numSteps;
+       weeklyAverage.minutesActive += day.minutesActive; 
+    }  
+       return weeklyAverage; 
+    }, {});
```
Comment (ConnorAndersonLarson) [inline src/Activity.js]: This reduce is doing a lot to make it work as a reduce. Thoughts for refactoring:

`const averageWeeklyData = {steps: 0, minutesActive: 0};`
 `currentWeekData.forEach(day => {`
`      averageWeeklyData.steps += day.numSteps;`
`      averageWeeklyData.minutesActive += day.minutesActive;`
`    });`

**Review by ConnorAndersonLarson — APPROVED** [review]
Overall looks great! Had some thoughts for sad Testing and a small refactor in comments, but everything works and can be pushed through!

File: src/Activity.js (hunk 2 above)
Comment (elizhahn) [reply src/Activity.js]: Oh nice, yeah I will implement this, I didn't like how big it was getting myself. Nice suggestion

File: test/Activity-test.js (hunk 1 above)
Comment (elizhahn) [reply test/Activity-test.js:58]: Good idea, let's implement this on a branch for sad Path testing additions

**Review by owenhallgren — APPROVED** [review]

**Pull request merged**